### PR TITLE
docs: user stage

### PR DESCRIPTION
**Pull request by soyeric128**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- added user stage;
- added SHOW STAGES;

Closes #issue

### CODE REVIEW

File: docs/doc/30-reference/30-sql/00-ddl/40-stage/index.md [hunk 1]
```diff
@@ -1,32 +1,64 @@
...
+
+### Named Internal / External Stage
+
+In addition to the [User Stage](#user-stage), you can create named stages to store data files. Named stages give you more control over the data loading:
+
+- Named stages enable other users with appropriate privileges to access the staged data files and load them into tables.
+
+- A named stage can be internal or external:
+
+    - Internal stages store data files in `/<bucket_name>/<tenant_id>/stage/internal/<stage_name>`.
```
Comment (Xuanwo) [inline docs/doc/30-reference/30-sql/00-ddl/40-stage/index.md:44]: The internal storage layout is by the implementation. It's better not to record in docs.

How about the following:

Internal stages store data files in the same storage backend of databend-query.

File: docs/doc/30-reference/30-sql/00-ddl/40-stage/06-ddl-show-stages.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+## Syntax
+
+```sql
+SHOW STAGES;
+```
+
+## Examples
+
+```sql
+SHOW STAGES;
```
Comment (Xuanwo) [inline docs/doc/30-reference/30-sql/00-ddl/40-stage/06-ddl-show-stages.md:17]: Maybe we need to show user stage too.

cc @BohuTANG @wubx for ideas.

Comment (BohuTANG) [reply docs/doc/30-reference/30-sql/00-ddl/40-stage/06-ddl-show-stages.md:17]: Indeed not. The stage is user level from the users' view.

File: docs/doc/30-reference/30-sql/00-ddl/40-stage/index.md [hunk 3]
```diff
@@ -41,7 +41,7 @@ In addition to the [User Stage](#user-stage), you can create named stages to sto
 
 - A named stage can be internal or external:
 
-    - Internal stages store data files in `/<bucket_name>/<tenant_id>/stage/internal/<stage_name>`.
+    - Internal stages store data files in the object storage you specify in `databend-query.toml`.
```
Comment (Xuanwo) [inline docs/doc/30-reference/30-sql/00-ddl/40-stage/index.md:44]:
```suggestion
    - Internal stages store data files in the storage backend you specify in `databend-query.toml`.
```

**Pull request merged**